src/details/ArborX_DetailsTreeConstruction.hpp

```diff
@@ -209,20 +136,42 @@ namespace
 int constexpr UNTOUCHED_NODE = -1;
 } // namespace
 
-template <typename MemorySpace>
-class GenerateHierarchyFunctor
+template <typename ExecutionSpace, typename Primitives, typename Nodes>
+inline void initializeSingleLeafNode(ExecutionSpace const &space,
```

Comment:
I'd like this to be moved above `UNTOUCHED_NODE`, as logically that one belongs next to `GenerateHierarchy`.